include/ginkgo/core/log/record.hpp

```diff
@@ -51,6 +51,8 @@ namespace log {
 
 /**
  * Struct representing iteration complete related data
+ *
+ * @ingroup log
```

Comment:
What if you add the `@ingroup` to the namespace instead? Does this get propagated to all internal structures? Here documenting the namespace could be useful simply for saving so many `@ingroup`s.